components/LinkCustom/LinkCustom.tsx

```diff
@@ -3,35 +3,71 @@
 
 import Link from 'next/link';
 import React, { JSX } from 'react';
+import { cva, type VariantProps } from 'class-variance-authority';
 import { cn } from '@/utils/utils';
 
-interface ILinkCustomProps {
-  children: React.ReactNode;
+const linkCustomVariants = cva(
+  'inline-flex items-center justify-center whitespace-nowrap rounded-md transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed',
+  {
+    variants: {
+      variant: {
+        default: 'underline underline-offset-4 hover:text-primary-muted transition-colors',
+        primaryButton: 'bg-primary text-primary-foreground hover:bg-primary-muted text-sm font-medium',
+        disabledPrimaryButton: 'bg-primary text-primary-foreground hover:bg-primary-muted text-sm font-medium opacity-50 cursor-not-allowed',
+        secondaryButton: 'bg-secondary text-secondary-foreground hover:bg-secondary-muted text-sm font-medium',
+        disabledSecondaryButton: 'bg-secondary text-secondary-foreground hover:bg-secondary-muted text-sm font-medium opacity-50 cursor-not-allowed',
+      },
+      size: {
+        default: 'h-fit w-fit',
+        defaultButton: 'h-10 px-4 py-2',
+        smButton: 'h-9 rounded-md px-3',
+        lgButton: 'h-11 rounded-md px-8',
+        icon: 'h-10 w-10',
+      }
+    },
+    defaultVariants: {
+      size: 'default',
+      variant: 'default',
+    },
+  }
+);
+
+interface ILinkCustomProps extends VariantProps<typeof linkCustomVariants> {
+  children?: React.ReactNode;
   className?: string;
+  dataTestidProp?: string;
   href: string;
+  // eslint-disable-next-line no-unused-vars
```

Comment:
As dicusssed with previous PR comments that contained this comment, please remove this as advised by Shashi.